subsys/logging/backends/log_backend_adsp_mtrace.c

```diff
@@ -115,7 +139,9 @@ static size_t mtrace_out(int8_t *str, size_t len, size_t *space_left)
 	if (space_left) {
 		*space_left = avail > len ? avail - len : 0;
 	}
-
+#if !defined(CONFIG_MM_DRV_INTEL_ADSP_MTL_TLB)
+	notify_observer(observer_dev, observer_slot_index, mtrace_data_buffer);
```

Comment:
Hmm, this would create a 4K memcpy for each log addition. This seems a bit too much overhead. Couldn't we have an option to get the buffer address from dynamic_slots.h, allowing to directly modify the slot memory. I get the benefit when mapping data that is anyways stored elsewhere and the mm remapping allows to avoid to same memory (and writes), but for this log data, there's no savings here and direct mode should be doable as well.

Reply:
And another issue is that the host will also modify the first bits of this slot and this observer construct will basicly not reflect the data host wrote (and the memcpy potentially overwrites anything the host writes).

Reply:
@kv2019i Hi, please check the new version. Dynamic slots allocation is still available for both platforms, but now you can  choose if you want to use TLB with mirroring or write directly to the memory as you were before for each slot separately. This should not affect legacy platform like TGL.

Reply:
Thanks @piotrkmx! Will check this and comment back here.